### DISCUSSION TITLE
Development Workflow

### DISCUSSION THREAD

**Discussion by azzamsa** [Orange-OpenSource/hurl#1112] [General] [answered]
Hi.

I am very glad this tool exists. I have two question regarding the development workflow.

- What is purpose of signed commit? Should we put this reason somewhere in the docs?
- Why hurl always need newest deps: [1](https://github.com/Orange-OpenSource/hurl/actions/runs/3716735257/jobs/6303402352#step:13:30), [2](https://github.com/Orange-OpenSource/hurl/blob/98d0299ff7013a0ccaf2ccb26d8e79e7c3f0a337/bin/update_crates.sh#L44).

Thanks a lot for Hurl! :heart:

**Comment by jcamiel** [comment, marked as answer]
Hi,

1 - the signed commit requirement is here to guarantee someone that a code has been changed by the author and not someone impersonating the author. With git, you can just set `git config user.name 'Harisson Ford'` and `git config user.email harisson.ford@gmail.com`. With the GPG signature, verified by GitHub, you've stronger guarantees on the author's identity. You can find a lot of resources on the web on the benefit of signing your commits. Many other repos require signed commit [Python](https://github.com/python/cpython), [Deno](https://github.com/denoland/deno) among others.

2 - We always update Hurl with the newest dependency as soon as possible:
  - this way, "major" upgrades are easier,
  - your dependencies are not "rotten": CVEs are fixed
  - Hurl benefits right way from bug resolutions and performance improvments.

We know the benefits and the drawbacks of dependencies and we prefer this way of managing dependencies. We see it as "we prefer to do a little easy cleaning every day than a big difficult cleaning every 6 months"!

**Reply by azzamsa** [reply]
Thank you so much for your detailed and thorough reply!
